xml2json.py

```diff
@@ -223,6 +224,11 @@ def validatelinks(root, node):
             'u', 'suffixes', 'extra', 'examples', 'footer']
 filenames = []
 concat=''
+
+argparser = argparse.ArgumentParser(description="Read the database XML files and output a JSON representation of the database to stdout, and report unresolvable links to stderr.")
+argparser.add_argument("--normalize", type=str, default="NFKD", choices=["NFC", "NFKD"], help="Which unicode normalization to use")
```

Comment:
The change description doesn't explain the motivation for adding support for the NFC normalization. What exactly is the reason this is needed? In other words, is NFKD unsuitable because of the K (compatibility normalization) or the D (decomposition)?

If the D is fine, and the problem is the K, then you can probably just change this to NFD and avoid having to add a new command line option. The decomposed normalization was added with 88e759c4f8edad2640632cfb1eeb208b9520eae3 to make it simpler to convert case for languages using accented latin characters. I went with the KD normalization in the hopes of avoiding issues with e.g. half width vs. full width CJK characters, but it's probably unlikely to actually be a problem in practice. If it ever does end up being a problem, I can handle it by doing the K normalization in the consumer of the JSON file rather than at generation time.

If this option is indeed needed, might as well add NFD and NFKC as valid options, for completeness.

Reply:
The reason is decomposition, as it doesn't work well with letters like äöü. My text editor cannot find words when I search with them because it thinks characters I inputted and the decomposed characters are different.